rustls/src/msgs/deframer.rs

```diff
@@ -491,6 +491,15 @@ fn copy_into_buffer(buf: &mut [u8], src: &[u8], at: usize) {
 }
 
 trait FilledDeframerBuffer {
+    fn filled_get_mut<I>(&mut self, index: I) -> &mut I::Output
```

Comment:
Style nit: `impl SliceIndex<[u8]>`?

Reply:
I think because of the `Output` associated type being referred to in the `return` using `impl` ends up being pretty ugly. The return needs to be qualified to avoid ambiguity:

```rust
fn filled_get_mut(
        &mut self,
        index: impl SliceIndex<[u8], Output = [u8]>,
    ) -> &mut <RangeFrom<usize> as SliceIndex<[u8]>>::Output 
```

I'm inclined to leave it as-is with the generic parameter. WDYT?

Reply:
Yeah, the impl argument doesn't make sense, sorry. I'd probably inline the I bound in the declaration instead of having a where clause, too.

Reply:
Fixed.